plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -270,80 +251,34 @@ public int getAdminPort() {
   /**
    * Starts the development server.
    *
-   * @param runnables the path to directories that contain configuration files such as
-   *        appengine-web.xml
    * @param console the stream (Eclipse console) to send development server process output to
    * @param arguments JVM arguments to pass to the dev server
    */
-  void startDevServer(List<File> runnables, MessageConsoleStream console, List<String> vmArguments)
+  void startDevServer(DefaultRunConfiguration devServerRunConfiguration,
+      MessageConsoleStream console)
       throws CoreException {
     
-    checkAndSetPorts();  // Must be called before setting the STARTING state.
-    setServerState(IServer.STATE_STARTING);
+    // Check ports before setting the STARTING state.
+    Predicate<Integer> portProber = new Predicate<Integer>() {
 
-    // Create dev app server instance
-    initializeDevServer(console);
+      @Override
+      public boolean apply(Integer port) {
```

Comment:
`port` seems never null. I think it's less confusing to have `Preconditions.checkNotNull(port)` rather than checking null here. One less thing to ever worry about.

After that, the name of the `Predicate` variable could be changed to `portInUse` from `portProber`.

I miss Java 8.